lib/hedwig_sms.ex

```diff
@@ -8,7 +8,7 @@ defmodule Hedwig.Adapters.SMS do
   @doc false
   def init({robot, opts}) do
     HTTPoison.start
-    Hedwig.Robot.register(robot, opts[:name])
+    :global.register_name(opts[:name], robot)
```

Comment:
Hey @bryanjos, yes...you should only register the adapter since the adapter will already keep track of the robot pid, you don't need to register it for yourself. So in your case, the plug function will need to find the adapter to send the message to, and the adapter process will just call `handle_in/2` on the robot pid.

However, you should still call `Hedwig.Robot.handle_connect/1` after the adapter has initialized so that the user has a chance to register their bot or do any additional things.

Sorry, I had a brain fart yesterday.